src/BSeries.jl

```diff
@@ -1453,4 +1456,420 @@ include("latexify.jl")
     include("precompile.jl")
 end
 
+
+"""
+        is_energy_preserving(rka::RungeKuttaMethod,order)::Bool
+
+This function checks whether a Runge-Kutta method `rk` is 
+energy-preserving for a given `order`.
+"""
+function is_energy_preserving(rk::RungeKuttaMethod,order)
+#generate bseries 
+    series = bseries(rk, order)
+    #pass it to the main functoin 'is_energy_preserving'
+    is_energy_preserving(series)
+end
+
+"""
+
+        is_energy_preserving(series)::Bool
+        
+        This function checks whether the B-series `series` of a time integration
+            method is energy-preserving for Hamiltonian systems - up to the
+            [`order`](@ref) of the `series`.
+            # References
+            This code is based on the Theorem 2 of
+            - Elena Celledoni, Robert I. McLachlan, Brynjulf Owren, and G. R. W. Quispel. 
+              "Energy-preserving integrators and the structure of B-series."
+              Foundations of Computational Mathematics 10 (2010): 673-693.
+              [DOI: 10.1007/s10208-010-9073-1](https://link.springer.com/article/10.1007/s10208-010-9073-1)
+ """
+function is_energy_preserving(series)
+    series_a = modified_equation(series)
+    #save all the coefficients in an array: it is easier
+    #to call an array full of fractions when working with the 
+    #functions 'renormalize_bseries' and 'energy_preserving_trees_test' 
+    coefficients = collect(values(series_a))
+    #save all the RootedTrees in another array: 
+    atrees = collect(keys(series_a))
+    #we need only the level sequence
+# Create an empty vector to store the converted trees into arrays
+    trees = Vector{Vector{Int}}(undef, length(series_a))
```

Comment:
Why do you work with a vector of level sequences instead of a vector of rooted trees?

Reply:
I chose to work with the level sequences because we have to modify the numbers inside a tree (for example, in the function `modify_t_sub`), and I found it easier by using arrays instead of Rooted Trees